File: WebGL/OGLRen/LineCell.js
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a college course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Computer Graphics 
//           Rensselaer Polytechnic Institute
//  Date:    Sept 2015
//
//////////////////////////////////////////////////////////////////////////

var OGLRen = OGLRen || {};

/**
  * LineCell class is a concrete implementation of the Cell class
  * It draws lines between point in the internal array, i.e.
  *   [ 1 2 2 3 3 4 4 5 ] will draw a line from 1-2-3-4-5
  *   [ 1 2 3 4 5 ] will draw a line from 1-2 then 3-4 skipping 5
  */
OGLRen.LineCell = function() {
    
	/**
	  * Sent by the Renderer to start the drawing process, kind of neat
	  * in that the LineCell sends commands back to the Renderer
	  * on how to draw itself (Remember the Renderer only has a vector
	  * of cells).
	  */
	this.render = function( aren)
	{
		var cnt = 0, ptsBuffer = null;

		cnt = this.getMaterials().size();
		aren.beginDraw(aren.LINES);
        if (this.initLess) {
            for (var i=0;i<this.size();i++) {
                aren.setMaterial(this.getMaterials().getMaterial(this.getVal(i)%cnt));
                aren.vertex(this.getPoints().getPoint(this.getVal(i))); 
            }
            ptsBuffer = aren.getPtsBuffer();
            this.initLess = false;
		}
		aren.endDraw(ptsBuffer);
	}
}
OGLRen.LineCell.inheritsFrom(OGLRen.Cell);

File: ClassPrevious/oglren/LineCell.cxx
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a graduate level course to show certain aspects of data
//  storage algorithms. It has problems and should not be used 
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Advanced Computer Graphics and Data visualization
//           Rensselaer Polytechnic Institute
//  Date:    October 3, 1995
//
//////////////////////////////////////////////////////////////////////////

#include "LineCell.h"

//
// Need to decide weather the display list needs to be created again
// If the mtset or ptset has been updated
void
LineCell::Render(Renderer *aren)
{
  int cnt;

	  cnt = this->MtSet->Count();
	  aren->BeginDraw(Renderer::LINE);
	  for (int i=0;i<this->Count();i++) {
	    aren->SetMaterial(this->MtSet->Get((this->Get(i))%cnt));
	    aren->Vertex(PtSet->Get(this->Get(i))->x, 
			PtSet->Get(this->Get(i))->y, 
			PtSet->Get(this->Get(i))->z);
	  }
	  aren->EndDraw();
}

void
LineCell::PrintSelf(ostream& os) {

  Cell::PrintSelf(os);
  os << "   Vertices: " << endl;
  for (int i=0;i<this->Count();i++) {
    os <<  "\t[" << PtSet->Get(this->Get(i))->x << "," << 
					PtSet->Get(this->Get(i))->y << "," <<
					PtSet->Get(this->Get(i))->z << "] " << endl;
  }
  os << endl;
}
